src/main/java/io/vertx/ext/hawkular/impl/Scheduler.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2015 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.hawkular.impl;
+
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.ext.hawkular.VertxHawkularOptions;
+import org.hawkular.metrics.client.common.SingleMetric;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static java.util.concurrent.TimeUnit.*;
+
+/**
+ * Collects metrics and relay them to the sender.
+ *
+ * @author Thomas Segismont
+ */
+public class Scheduler {
+  private final Vertx vertx;
+  private final Handler<List<SingleMetric>> sender;
+  private final List<MetricSupplier> suppliers;
+
+  private long timerId;
```

Comment:
Right, but the thread which will call stop will do it just once and can't use a cached value, can it?